blog/content/post/2020-12-14-trenchboot_attestation.md

```diff
@@ -0,0 +1,266 @@
+Authentication Secrets IDs specify the target environment, which must provide
+the evidence. Claim selection tells the attester which claims should be
+included in the evidence.
+
+In the next step, the attester collects the selected claims and on this base, it
+generates the evidence. The attester sends to the verifier the message consists
+of the evidence and the event log. The verifier appraises the evidence and it
+creates the attestation result, which should be passed to the Relying Party.
+
+More information about the Reference Interaction Models for Remote Attestation
+Procedures is presented in the IETF
+[document](https://www.ietf.org/archive/id/draft-birkholz-rats-reference-interaction-model-03.txt).
+
+
+## Proof of Concept Implementation of the CHARRA
+
+The Fraunhofer SIT provides the [proof-of-concept implementation](https://github.com/Fraunhofer-SIT/charra)
+for CHARRA. The Verifier and the attester are two separate C scripts
+that provide functionalities that were described above. The
+proof-of-concept implementation assumes that the verifier and the attester
+are running in the same docker container. Though we cannot say that is the remote
+attestation. The verifier and attester are using the TPM Simulator instead of
+a physical one.
+
+Our goal is to provide the CHARRA for physical TPM and separate the attester
+and verifier.
+
+### Separation of the attester and verifier
+
+As I mentioned before the proof of concept (PoC) implementation of the CHARRA
+uses the docker container to provide and appraise evidence. Attestation
+data is obtained via a TPM Quote function. It provides the quote and signature
+for the given list of PCRs. In the PoC case, the attester and verifier generate
+separates keys. In that case verification of TPM quote and signature will
+work only, when the keys are generated by the same TPM device. Otherwise,
+the evidence verification will fail due to a mismatch of the attestation
+identity key. To separate the instances we need to verify the signature and
+attestation data with the attestation public key.
+
+In the PoC implementation, there is no endorser and key registration system.
+Every time the verifier sends the attestation request, the attester is
+generating the new attestation key based on the nonce. We need to obtain and
+send to the verifier TPM public key that is used to generate the TPM quote.
+TPM API (Esys) during the key creation allows obtaining the public and private
+part of the attestation key. We added to the `charra_key_mgr` the additional
+parameter that conveys the public key to the attester.
+
+The communication between attester and verifier is provided by libcoap library.
+It is the C implementation of the Constrained Application Protocol.
+Currently, the communication between the attester and verifier is constrained by
+a maximum transmission unit (~1500 bytes). The block-wise transmission is in the
+development plans. Though we need to fit the public key with the TPM signature
```

Comment:
This limitation was introduced by the PoC from Fraunhofer SIT? This improvement is on their roadmap or on ours?

Reply:
@macpijan 

> This limitation was introduced by the PoC from Fraunhofer SIT?

Yes, it was.

>  This improvement is on their roadmap or on ours?

It was on their roadmap.